src/spinegeneric/cli/check_data_consistency.py

```diff
@@ -99,6 +99,14 @@ def main():
     parser = get_parser()
     args = parser.parse_args()
 
+    # Keep track of errors for the exit status
+    warnings_found = False
+
+    def warn(message: str):
+        nonlocal warnings_found
+        warnings_found = True
+        print(f"Warning: {message}")
```

Comment:
I imagine trying to deduplicate this function and dealing with module-level scope would be too messy to be worth it? :stuck_out_tongue_closed_eyes: